doc/BUILD.rst

```diff
@@ -0,0 +1,730 @@
+``mingw64`` cross-compiler toolchains available as well. That's is just the one
+that works for us.
+
+Additionally, you can install `nsis <https://nsis.sourceforge.io/Main_Page>`__
+(>= 3.0) for packaging Windows installer ``.exe`` files. (However, you can
+still package ``.zip`` files for Windows without it.)
+
+On macOS, you're probably better off using Docker and the
+`Docker container <https://hub.docker.com/r/mstorsjo/llvm-mingw/>`__ that
+``llvm-mingw`` provides, and installing ``nsis`` on top of it.
+
+Another options for Windows-based computers is leveraging Windows
+10's
+`Windows For Linux (WSL) Subsystem <https://docs.microsoft.com/en-us/windows/wsl/install-win10>`__
+to cross-compile to Windows using their Ubuntu image. You can also potentially
+use a ``mingw64`` toolchain directly on Windows, however that isn't supported
+or recommended, as it's generally more trouble than its worth.
+
+Checking Out Code
+-----------------
+
+First, you'll need to checkout the repository. You can do that with the
+following:
+
+.. code:: sh
+
+   git clone --recurse-submodules https://github.com/taisei-project/taisei
+
+The ``git submodule update --init --recursive`` line is absolutely necessary,
+or Taisei will not build, as it will be missing many of the dependencies its
+needs to compile.
+
+Build Options
+-------------
+
+This is *not* an exhaustive list. You can see the full list of option using
+``meson`` in the ``taisei`` directory.
+
+.. code:: sh
+
+   cd taisei/
+   meson configure
+
+Setup
+"""""
+
+The first command you'll need to run is ``setup``, which creates a directory
+(in this case, ``taisei/build/``). It checks your system for various
+dependencies and required tools, which should take about a minute on most
+systems.
+
+.. code:: sh
+
+   # inside the taisei/ directory you cloned before
+   meson setup -C build/
+
+You can also have the ``setup`` command contain certain build options (seen
+below). The following are an *example* and *not required* for getting Taisei
+building.
+
+.. code:: sh
+
+   # enables Developer Mode and debugging symbols
+   meson setup -C build/ -Ddeveloper=true -Dbuildtype=debug
+
+You can then apply more build options later using ``meson configure`` (as seen
+below). It will automatically reconfigure your build environment with the new
+options without having to rebuild everything.
+
+System/Vendored Dependencies (``--wrap-mode``)
+""""""""""""""""""""""""""""""""""""""""""""""
+
+See: `Meson Manual <https://mesonbuild.com/Wrap-dependency-system-manual.html>`__
+
+* Default: ``default``
+* Options: ``default``, ``nofallback``, ``forcefallback``, ...
+
+This is a core ``meson`` flag that does quite a few things. Not all of them will
+be covered here. Refer to the ``meson`` documentation linked above.
+
+Generally, ``default`` will rely on system-installed libraries when available,
+and fallback to vendored in-repository dependencies when necessary.
```

Comment:
It's more accurate to say that it'll download the missing dependencies, as we don't vendor stuff in-tree with wraps.